packages/components/mention/src/mention.vue

```diff
@@ -0,0 +1,250 @@
+const tooltipRef = ref<TooltipInstance>()
+const dropdownRef = ref<InstanceType<typeof ElMentionDropdown>>()
+
+const visible = ref(false)
+const cursorStyle = ref<CSSProperties>()
+const mentionCtx = ref<MentionCtx>()
+
+const computedPlacement = computed<Placement>(() =>
+  props.showArrow ? props.placement : `${props.placement}-start`
+)
+
+const computedFallbackPlacements = computed<Placement[]>(() =>
+  props.showArrow ? ['bottom', 'top'] : ['bottom-start', 'top-start']
+)
+
+const filteredOptions = computed(() => {
+  const { filterOption, options } = props
+  if (!mentionCtx.value || !filterOption) return options
+  return options.filter((option) =>
+    filterOption(mentionCtx.value!.pattern, option)
+  )
+})
+
+const handleInputChange = (value: string) => {
+  emit('update:modelValue', value)
+  syncAfterCursorMove()
+}
+
+const handleInputKeyDown = (e: KeyboardEvent | Event) => {
+  if (!('key' in e)) return
+  if (elInputRef.value?.isComposing) return
+  if (['ArrowLeft', 'ArrowRight'].includes(e.key)) {
+    syncAfterCursorMove()
+  } else if (['ArrowUp', 'ArrowDown'].includes(e.key)) {
+    if (!visible.value) return
+    e.preventDefault()
+    const direction = e.key === 'ArrowUp' ? 'prev' : 'next'
+    dropdownRef.value?.navigateOptions(direction)
+  } else if (['Enter'].includes(e.key)) {
+    if (!visible.value) return
+    e.preventDefault()
+    if (dropdownRef.value?.hoverOption) {
+      dropdownRef.value?.selectHoverOption()
+    } else {
+      visible.value = false
+    }
+  } else if (['Backspace'].includes(e.key)) {
+    if (props.whole && mentionCtx.value) {
+      const { splitIndex, selectionEnd, pattern, prefixIndex, prefix } =
+        mentionCtx.value
+      const inputEl = getInputEl()
+      if (!inputEl) return
+      const inputValue = inputEl.value
+      const matchOption = props.options.find((item) => item.value === pattern)
+      const isWhole = isFunction(props.checkIsWhole)
+        ? props.checkIsWhole(pattern, prefix)
+        : matchOption
+      if (isWhole && splitIndex !== -1 && splitIndex + 1 === selectionEnd) {
+        e.preventDefault()
+        const newValue =
+          inputValue.slice(0, prefixIndex) + inputValue.slice(splitIndex + 1)
+        emit(UPDATE_MODEL_EVENT, newValue)
+
+        const newSelectionEnd = prefixIndex
+        nextTick(() => {
+          // input value is updated
+          inputEl.selectionStart = newSelectionEnd
+          inputEl.selectionEnd = newSelectionEnd
+          syncDropdownVisible()
+        })
+      }
+    }
+  }
+}
+
+let blurTimer: ReturnType<typeof setTimeout> | undefined
+
+const handleInputFocus = (e: FocusEvent) => {
+  if (blurTimer) clearTimeout(blurTimer)
+  emit('focus', e)
+  syncAfterCursorMove()
+}
+
+const handleInputBlur = (e: FocusEvent) => {
+  emit('blur', e)
+  // the blur event fires faster than the click event, so a delay is needed here
+  blurTimer = setTimeout(() => {
+    visible.value = false
+  }, 200)
+}
```

Comment:
```suggestion
const { wrapperRef, handleFocus, handleBlur } = useFocusController(elInputRef, {
  afterFocus() {
    syncAfterCursorMove()
  },
  beforeBlur(event) {
    return tooltipRef.value?.isFocusInsideContent(event)
  },
  afterBlur() {
    visible.value = false
  },
})
```

```diff
- <div :class="ns.b()">
+ <div ref="wrapperRef" :class="ns.b()">

- @focus="handleInputFocus"
- @blur="handleInputBlur"
+ @focus="handleFocus"
+ @blur="handleBlur"
```

// hooks/use-focus-controller/index

```diff
- export function useFocusController<T extends HTMLElement>(
+ export function useFocusController<T extends { focus: () => void }>(
```

I optimized the code, I wonder if it meets your needs?

This way the component will emit correct focus and blur events.

Reply:
thanks, It solves the problem.